contactaccessrights/js/src/access-rights/controllers/access-rights-modal-ctrl.js

```diff
@@ -0,0 +1,113 @@
+define(['access-rights/modules/controllers'], function (controllers) {
+  'use strict';
+
+  controllers.controller('AccessRightsModalCtrl', ['Region', 'Location', 'Right', '$q', '$modalInstance',
+    function (Region, Location, Right, $q, $modalInstance) {
+      var vm = this;
+
+      vm.availableData = {
+        regions: [],
+        locations: []
+      };
+
+      vm.selectedData = {
+        locations: [],
+        regions: []
+      };
+
+      vm.originalData = {
+        locations: [],
+        regions: []
+      };
+
+      vm.errorMsg = '';
+
+      /**
+       * Closes the modal
+       */
+      vm.cancel = function () {
+        $modalInstance.dismiss('cancel');
+      };
+
+      /**
+       * Saves data and closes the modal
+       */
+      vm.submit = function () {
+        $q.all([persistRegions(), persistLocations()])
+          .then(function () {
+            $modalInstance.dismiss('cancel');
+          })
+          .catch(function (err) {
+            vm.errorMsg = 'Error while saving data';
+          });
+      };
+
+      /**
+       * Saves the new regions, and deletes the removed ones
+       *
+       * @return {Promise}
+       */
+      function persistRegions() {
+        return persistValues(vm.originalData.regions, vm.selectedData.regions,
+          Right.saveRegions.bind(Right));
+      }
+
+      /**
+       * Saves the new locations, and deletes the removed ones
+       *
+       * @return {Promise}
+       */
+      function persistLocations() {
+        return persistValues(vm.originalData.locations, vm.selectedData.locations,
+          Right.saveLocations.bind(Right));
```

Comment:
Given that the only difference in `persistValues` is whether you are dealing with regions or locations, why not something like:

```js
/**
 * Saves the new values, and deletes the removed ones
 *
 * @param  {string} type - either 'regions' or 'locations'
 * @return {Promise}               The result of all promises
 */
function persistValues(type) {
  var originalData = vm.originalData[type];
  var selectedData = vm.selectedData[type];

  // ....

  if (newEntityIds.length > 0)
    promises.push(Right['save' + _.upperFirst(type)](newEntityIds));

  if (removedRightIds.length > 0)
    promises.push(Right.deleteByIds(removedRightIds));

  return $q.all(promises);
}


// Usage
/**
 * Saves data and closes the modal
 */
vm.submit = function () {
  // persistRegions and persistLocations are not needed anymore
  $q.all([persistValues('regions'), persistValues('locations')])
  // ...
};
```
this way the function's signature is simplified, you can scrap two private functions and you can avoid passing a binded method as a callback